### PR TITLE
Add json_dto/0.2.11 

### PR DESCRIPTION
**Pull request by ngrodzitski**
Add json_dto library of v.0.2.11 to conan-index (https://github.com/Stiffstream/json_dto).

Specify library name and version:  **json_dto/0.2.11**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/json_dto/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["JSON_DTO_INSTALL"] = True
+        self._cmake.definitions["JSON_DTO_FIND_DEPS"] = False
+        self._cmake.configure(source_folder=self._source_subfolder+"/dev/json_dto")
```
Comment (theirix) [inline recipes/json_dto/all/conanfile.py:59]: use `os.path.join` for portability

File: recipes/json_dto/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class JsondtoConan(ConanFile):
+    name = "json_dto"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/Stiffstream/json_dto"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A small header-only helper for converting data between json representation and c++ structs"
+    topics = ("JSON", "DTO", "Serialization")
```
Comment (theirix) [inline recipes/json_dto/all/conanfile.py:11]: Please downcase keywords

File: recipes/json_dto/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib"))
+
+    def package_id(self):
+        self.info.settings.clear()
```
Comment (theirix) [inline recipes/json_dto/all/conanfile.py:75]: [Use](https://docs.conan.io/en/latest/howtos/header_only.html) `self.info.header_only()`

File: recipes/json_dto/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    def package_info(self):
```
Comment (theirix) [inline recipes/json_dto/all/conanfile.py:77]: Should we export defines `RAPIDJSON_HAS_STDSTRING` and `RAPIDJSON_HAS_CXX11_RVALUE_REFS` as suggested in [upstream doc](https://github.com/Stiffstream/json_dto#how-to-use-it) ?

File: recipes/json_dto/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/json_dto/all/conanfile.py:59]: It's from me too 🤣 https://github.com/Stiffstream/json_dto-conan/pull/1

File: recipes/json_dto/all/conanfile.py (hunk 4 above)
Comment (ngrodzitski) [reply recipes/json_dto/all/conanfile.py:77]: Those defines are subject of rapidjson and the package for rapidjson has no options to set them. So json_dto makes no choices out of its own lib and uses rapidjson as is. There was an issue with using `std::string` with rapidjson API (if `RAPIDJSON_HAS_STDSTRING` is not defined), but currently json_dto avoids using `std::string` calling rapidjson functions so it is up to use to adjust rapidjson in this aspect.

Comment (theirix) [reply recipes/json_dto/all/conanfile.py:77]: Seems it works fine

**Review by theirix — APPROVED** [review]

File: recipes/json_dto/all/conanfile.py [hunk 5]
```diff
@@ -56,7 +56,7 @@ def _configure_cmake(self):
         self._cmake = CMake(self)
         self._cmake.definitions["JSON_DTO_INSTALL"] = True
         self._cmake.definitions["JSON_DTO_FIND_DEPS"] = False
-        self._cmake.configure(source_folder=self._source_subfolder+"/dev/json_dto")
+        self._cmake.configure(source_folder=os.path.join(self._source_subfolder, "dev/json_dto"))
```
Comment (prince-chrismc) [inline recipes/json_dto/all/conanfile.py:59]:
```suggestion
        self._cmake.configure(source_folder=os.path.join(self._source_subfolder, "dev", "json_dto"))
```

File: recipes/json_dto/all/conanfile.py [hunk 6]
```diff
@@ -72,7 +72,7 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib"))
 
     def package_id(self):
-        self.info.settings.clear()
+        self.info.header_only()
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "json_dto"
```
Comment (prince-chrismc) [inline recipes/json_dto/all/conanfile.py:78]: I believe this can be removed since it's the default

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**